casper/src/main/scala/coop/rchain/casper/merging/ConflictSetMerger.scala

```diff
@@ -124,27 +124,16 @@ object ConflictSetMerger {
     /** target function for rejection is minimising cost of deploys rejected */
     val rejectionTargetF = (dc: Branch) => dc.map(cost).sum
 
+    val rejectionOptionsWithOverflow = getMergedResultRejection(
+      branches,
+      rejectionOptions,
+      baseMergeableChRes
+    )
+    val optimalRejection = getOptimalRejection(rejectionOptionsWithOverflow, rejectionTargetF)
+    val rejected         = lateSet ++ rejectedAsDependents ++ optimalRejection.flatten
+    val toMerge          = branches diff optimalRejection
     for {
-      baseMergeableChRes <- branches.flatten
```

Comment:
This reads init values in mergeable channels for all event log indices in the conflict scope. So it can be done outside of the function, before running conflict resolution.